packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts

```diff
@@ -425,6 +432,64 @@ export class StripeWebhookHandler extends StripeHandler {
     await this.updateCustomer(uid, email);
   }
 
+  /**
+   * Validate plan metadata on update
+   */
+  async handlePlanUpdatedEvent(request: AuthRequest, event: Stripe.Event) {
+    const plan = event.data.object as Stripe.Plan;
+    const products = await this.stripeHelper.fetchAllProducts();
```

Comment:
Since this is a plan update, we can used the cached products from `allProducts`.

Reply:
When I wrote the comment above I thought this patch was going to invalidate the products cache on the product updated event handler; I noticed that it is not.  In that case, we should fetch just one product here instead of all of them, since we have the product id.